coding/reader.hpp

```diff
@@ -19,10 +20,12 @@ class Reader
   DECLARE_EXCEPTION(SizeException, Exception);
   DECLARE_EXCEPTION(ReadException, Exception);
 
+  using TReaderPtr = unique_ptr<Reader>;
```

Comment:
TReaderUniquePtr is almost the same as unique_ptr<Reader>. What's the difference? Why not to use unique_ptr<Reader>? TReaderUniquePtr is as ugly as unique_ptr<Reader>, so how about to use latter?